libcontainerd/remote_daemon.go

```diff
@@ -263,11 +263,9 @@ func (r *remote) startContainerd() error {
 func (r *remote) monitorConnection(monitor *containerd.Client) {
 	var transientFailureCount = 0
 
-	ticker := time.NewTicker(500 * time.Millisecond)
-	defer ticker.Stop()
-
 	for {
-		<-ticker.C
+		time.Sleep(500 * time.Millisecond)
```

Comment:
Shouldn't this do a select with `shutdownContext.Done()` instead?

Reply:
Not really related to this fix though. In that case, you would still replace the ticker with `time.After`.

Reply:
Updated